# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
# Commit: 3d0e6a5fdfb9
# Full Hash: 3d0e6a5fdfb9f2a0618e4dcd8637b2369259f09e
# Author: serge-sans-paille <sguelton@mozilla.com>
# Date: 2024-11-06 16:51:31
# Regressor Bug: 1929213
# File Overlap Count: 1
# Description:
#   Bug 1929213 - Use std::clamp instead of min/max chain in dom/media/webrtc r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227936
# ==============================================================================

diff -r 66fa16b4e99e -r 3d0e6a5fdfb9 dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
--- a/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Wed Nov 06 10:14:24 2024 +0000
+++ b/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Wed Nov 06 10:15:25 2024 +0000
@@ -845,13 +845,13 @@
       int32_t prefHeight = aPrefs.GetHeight(isHd);
 
       cap.width = c.mWidth.Get(prefWidth);
-      cap.width = std::max(0, std::min(cap.width, 7680));
+      cap.width = std::clamp(cap.width, 0, 7680);
 
       cap.height = c.mHeight.Get(prefHeight);
-      cap.height = std::max(0, std::min(cap.height, 4320));
+      cap.height = std::clamp(cap.height, 0, 4320);
 
       cap.maxFPS = c.mFrameRate.Get(aPrefs.mFPS);
-      cap.maxFPS = std::max(0, std::min(cap.maxFPS, 480));
+      cap.maxFPS = std::clamp(cap.maxFPS, 0, 480);
 
       if (cap.width != prefWidth) {
         // Width was affected by constraints.